src/repository/postRepository.ts

```diff
@@ -4,17 +4,27 @@ import { db } from '../utils';
 
 class PostRepository {
 
+  // Selects the specifically wanted post by its id if found.
   async selectById(id: string): Promise<Post | undefined> {
     const query = 'select * from posts where id = ?';
     const [rows] = await db.pool.execute<RowDataPacket[]>(query, [id]);
-    if (rows[0]) return new Post(rows[0] as Post);
+    if (rows.length > 0) return new Post(rows[0] as Post);
   }
 
+  // Selects max 10 most recent posts. Where post's createdAt <= lastDate.
+  async selectRecent(lastDate: Date): Promise<Post[] | undefined> {
+    const query = 'select * from posts where createdAt <= ? order by createdAt desc limit 10';
```

Comment:
Let's use `createdAt < ?` instead of `createdAt <= ?`, since currently if we have posts (from least recent to most recent): 1,2,3...,29,30, then on the frontend first we would get posts 30-21, then 21-12, 12-3, 3-1, and 21, 12 and 3 would be returned twice. If we use `createdAt < ?` then we would get 30-21, 20-11, 10-1.